shared/qcommon/q_platform.h

```diff
@@ -229,13 +229,17 @@ along with this program; if not, see <http://www.gnu.org/licenses/>.
 // endianness
 // Use compiler builtins where possible for maximum performance
 #include <stdint.h>
-#if !defined(__clang__) && (defined(__GNUC__) || defined(__GNUG__))
-
-// gcc
+#if !defined(__clang__) && (defined(__GNUC__) || defined(__GNUG__)) \
+            && ((__GNUC__ * 100 + __GNUC_MINOR__) >= 403)
```

Comment:
I actually think the original way this was done was the right way. By changing the code to `__GNUC__ >= 4 && __GNUC_MINOR_C >= 3`, the check would fail for versions 5.0-5.2, 6.0-6.2, etc.